docs/architecture/adr-030-rosetta-api-support.md

```diff
@@ -0,0 +1,223 @@
+	Adapter    rosetta.Adapter
+}
+````
+
+A `NetworkProperties` `struct` comprises basic values that are required by a Rosetta API `Service` to run:
+
+```
+type NetworkProperties struct {
+	// Mandatory properties
+	Blockchain          string
+	Network             string
+	AddrPrefix          string
+	SupportedOperations []string
+}
+```
+
+Rosetta API services use Blockchain and Network as identifiers, e.g. the developers of gaia, the application that powers the Cosmos Hub, may want to set those to Cosmos Hub and cosmos-hub-3 respectively.
+
+`AddrPrefix` contains the network-specific address prefix. Cosmos SDK base implementations will default to `cosmos`, client applications are instructed that this should be changed according to their network configuration.
+
+`SupportedOperations` contains the transaction types that are supported by the library. At the present time, only `Transfer` is supported.
+
+`Network` holds an `Adapter` reference too. Adapter implementations may vary across different Cosmos SDK release series:
+
+```
+type Adapter interface {
+	DataAPI
+	ConstructionAPI
+}
+
+type DataAPI interface {
+	server.NetworkAPIServicer
+	server.AccountAPIServicer
+	server.MempoolAPIServicer
+	server.BlockAPIServicer
+	server.ConstructionAPIServicer
+}
+
+type ConstructionAPI interface {
+	server.ConstructionAPIServicer
+}
+
+````
+
+### Cosmos SDK Integration
+
+The best way to do it would be that every Cosmos SDK version holds the implementation of the Network struct, this can be an example located in the Cosmos SDK Launchpad version.
+
+```
+// NewLaunchpadNetwork returns a configured network to work in a Launchpad version.
+func NewLaunchpadNetwork(options Options) service.Network {
+	cosmosClient := cosmos.NewClient(fmt.Sprintf("http://%s", options.CosmosEndpoint))
+	tendermintClient := tendermint.NewClient(fmt.Sprintf("http://%s", options.TendermintEndpoint))
+
+	return service.Network{
+		Properties: rosetta.NetworkProperties{
+			Blockchain:          options.Blockchain,
+			Network:             options.Network,
+			SupportedOperations: []string{OperationTransfer},
+		},
+		Adapter: newAdapter(
+			cosmosClient,
+			tendermintClient,
+			properties{
+				Blockchain:  options.Blockchain,
+				Network:     options.Network,
+				AddrPrefix:  options.AddrPrefix,
+				OfflineMode: options.OfflineMode,
+			},
+		),
+	}
+}
+```
+
+As we see we hold not only the instantiation of a Launchpad specific network but the adapter implementation too. As the way to talk to different rpc endpoint change between different versions (Launchpad, Stargate, etc), makes sense to keep it in their respective branches.
+
+In this way we could have two options (yet to decide) that we can expose rosetta in the cosmos SDK.
+
+#### Embedding it in the Start command.
+
+This will look more like the approach the API in cosmos works: [link](https://github.com/cosmos/cosmos-sdk/blob/ca7b31dd5d23b10a919de1a60bbac90e89c5bf78/server/start.go#L248).
```

Comment:
`Rosetta API` service could run within the same execution process of the application; new configuration option and command line flag would be provided: